bazel/cross_platform.bzl

```diff
@@ -0,0 +1,20 @@
+def cc_posix_binary(name, srcs = [], **kargs):
```

Comment:
@rnburn the CI is failing because [the way that envoy does distro-specific libraries](https://github.com/envoyproxy/envoy/blob/master/bazel/envoy_library.bzl#L74-L101) doesn't translate to building binaries. Since the cc_binary rules is expected to produce outputs, `bazel build //...` still fails because, for example, `cc_windows_binary` produces no outputs on a mac. 

I'm looking for more advice regarding how to solve this problem.